File: sdk/c#/PicarioXPO.RenderAPI/XpoUrlDesign.cs
using System;

namespace PicarioXPO.RenderAPI
{
    /// <summary>
    /// Represents a design from a Picario scene for the XPO URL generator
    /// </summary>
    public sealed class XpoUrlDesign
    {
        /// <summary>
        /// Gets or sets the index of this overlay
        /// </summary>
        public int Index { get; set; }

        /// <summary>
        /// Gets or sets the index of this overlay
        /// </summary>
        public int SameIndex { get; set; }

        /// <summary>
        /// Gets or sets the type for this object
        /// </summary>
        public XpoUrlObjectTypes ObjectType { get; set; }

        /// <summary>
        /// Gets or sets the filename for this object
        /// </summary>
        public string EntityName { get; set; }

        /// <summary>
        /// Gets or sets the width of this object
        /// </summary>
        public double Width { get; set; }

        /// <summary>
        /// Gets or sets the height of this object
        /// </summary>
        public double Height { get; set; }

        /// <summary>
        /// Gets or sets the gloss of this object
        /// </summary>
        public double Gloss { get; set; }

        /// <summary>
        /// Gets of sets the contrast of this object
        /// </summary>
        public double Contrast { get; set; }

        /// <summary>
        /// Gets or sets the drop x of this object
        /// </summary>
        public double DropX { get; set; }

        /// <summary>
        /// Gets or sets the drop y of this object
        /// </summary>
        public double DropY { get; set; }

        /// <summary>
        /// Gets or sets the placing point x of this object
        /// </summary>
        public double PlacingPointX { get; set; }

        /// <summary>
        /// Gets or sets the placing point y of this object
        /// </summary>
        public double PlacingPointY { get; set; }

        /// <summary>
        /// Gets or sets the transformation for this object
        /// </summary>
        [Obsolete("No longer used in version 2. Use the Rotation and Flip parameters.")]
        public XpoUrlObjectTransformations Transformation { get; set; }
        
        /// <summary>
        /// Gets or sets the rotation for this object
        /// </summary>
        public int Rotation { get; set; }

        /// <summary>
        /// Gets or sets whether the texture should be flipped
        /// </summary>
        public bool Flip { get; set; }

        /// <summary>
        /// Specifies whether the design should repeat itself throughout the object.
        /// </summary>
        public bool Repeat { get; set; }
    }
}

File: sdk/js/PicarioXPO.RenderAPI.V2/UrlGenerators/XpoUrlDesign.ts
class XpoUrlDesign {
    private index: number;
    private sameIndex: number;
    private objectType: UrlGeneratorModule.XpoUrlObjectTypes;
    private entityName: string;
    private width: number;
    private height: number;
    private gloss: number;
    private contrast: number;
    private dropX: number;
    private dropY: number;
    private placingPointX: number;
    private placingPointY: number;
    private rotation: number;
    private flip: boolean;
    private repeat: boolean;
    private aspectRatio: boolean;

    constructor(fileName: string = "") {
        this.entityName = fileName;
    }

    /*
     * Gets or sets the index of this design
     */
    getIndex(): number { return this.index; }
    setIndex(val: number) { this.index = val; }

    /*
     * Gets or sets the index of this design
     */
    getSameIndex(): number { return this.sameIndex; }
    setSameIndex(val: number) { this.sameIndex = val; }

    /*
     * Gets or sets the type for this object
     */
    getObjectType(): UrlGeneratorModule.XpoUrlObjectTypes { return this.objectType; }
    setObjectType(val: UrlGeneratorModule.XpoUrlObjectTypes) { this.objectType = val; }

    /*
     * Gets or sets the filename for this object
     */
    getEntityName(): string { return this.entityName; }
    setEntityName(val: string) { this.entityName = val; }
    
    /*
     * Gets or sets the width of this object
     */
    getWidth(): number { return this.width; }
    setWidth(val: number) { this.width = val; }
    
    /*
     * Gets or sets the height of this object
     */
    getHeight(): number { return this.height; }
    setHeight(val: number) { this.height = val; }
    
    /*
     * Gets or sets the gloss of this object
     */
    getGloss(): number { return this.gloss; }
    setGloss(val: number) { this.gloss = val; }
    
    /*
     * Gets of sets the contrast of this object
     */
    getContrast(): number { return this.contrast; }
    setContrast(val: number) { this.contrast = val; }
    
    /*
     * Gets or sets the drop x of this object
     */
    getDropX(): number { return this.dropX; }
    setDropX(val: number) { this.dropX = val; }
    
    /*
     * Gets or sets the drop y of this object
     */
    getDropY(): number { return this.dropY; }
    setDropY(val: number) { this.dropY = val; }
    
    /*
     * Gets or sets the placing point x of this object
     */
    getPlacingPointX(): number { return this.placingPointX; }
    setPlacingPointX(val: number) { this.placingPointX = val; }
    
    /*
     * Gets or sets the placing point y of this object
     */
    getPlacingPointY(): number { return this.placingPointY; }
    setPlacingPointY(val: number) { this.placingPointY = val; }
    
    /*
     * Gets or sets the rotation for this object
     */
    getRotation(): number { return this.rotation; }
    setRotation(val: number) { this.rotation = val; }
    
    /*
     * Gets or sets whether the texture should be flipped
     */
    getFlip(): boolean { return this.flip; }
    setFlip(val: boolean) { this.flip = val; }
    
    /*
     * Specifies whether the design should repeat itself throughout the object.
     */
    getRepeat(): boolean { return this.repeat; }
    setRepeat(val: boolean) { this.repeat = val; }

    /*
     * Specifies whether the design should be rendered once while preserving the aspect ratio.
     */
    getAspectRatio(): boolean { return this.aspectRatio; }
    setAspectRatio(val: boolean) { this.aspectRatio = val; }
}